tools/structwrite/testdata/src/github.com/username/linttestmodule/structwrite.go

```diff
@@ -0,0 +1,114 @@
+// Package linttestmodule is a collection of code representing test cases for the linter.
+// See package golang.org/x/tools/go/analysis/analysistest for details.
+package linttestmodule
+
+// NonWritable is configured for linting.
+type NonWritable struct {
+	A int
+}
+
+func NewNonWritable() *NonWritable {
+	return &NonWritable{
+		A: 1,
+	}
+}
+
+func (nw *NonWritable) SetA() {
+	nw.A = 1 // want "write to NonWritable field outside constructor"
+}
+
+func NonWritableConstructLiteral() {
+	nw := NonWritable{}      // want "construction of NonWritable outside constructor"
+	nw = NonWritable{A: 1}   // want "construction of NonWritable outside constructor"
+	nwp := &NonWritable{}    // want "construction of NonWritable outside constructor"
+	nwp = &NonWritable{A: 1} // want "construction of NonWritable outside constructor"
+	//nwp = new(NonWritable)   // TODO - currently disallowed
+	var nwnw NonWritable // TODO - currently allowed
```

Comment:
We might want to allow this in the future as well, because [the storage layer uses this pattern to decode data into a provided struct.](https://github.com/onflow/flow-go/blob/master/storage/store/results.go#L29)
```suggestion
	var nwnw NonWritable
```

Reply:
This is a very good point. In general I'm leaning toward changing the linter to allow all empty initializations.

Reply:
would this pattern be allowed? It seems like that's still a modification which should get flagged. I wonder if we would need some special approach for deserializing from storage/grpc/network inputs.

Reply:
I modified the behaviour to allow empty construction based on the conversation here. Documentation is updated here: https://github.com/onflow/flow-go/pull/7310/files#diff-df50c4899b1555225511613798c685d9cf2bd8375b65b5204294efba8186c54dR24-R62